CMakeLists.txt

```diff
@@ -291,44 +291,102 @@ configure_file(
+  else ()
+    set(ARANGODB_BUILD_REPOSITORY "GIT FAILED TO RETRIEVE THE VERSION - UNSUPPORTED BUILD MODE")
+    set(HAVE_ARANGODB_BUILD_REPOSITORY "1")
+  endif()
+else()
   set(ARANGODB_BUILD_REPOSITORY "")
   set(HAVE_ARANGODB_BUILD_REPOSITORY "0")
 endif()
 
 if (DEFINED GIT_EXE AND USE_ENTERPRISE AND IS_DIRECTORY "${CMAKE_SOURCE_DIR}/enterprise/.git")
+  # Get enterprise commit hash 
   execute_process(
     WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/enterprise
-
-    COMMAND ${GIT_EXE} describe --all --tags --long --dirty=-dirty
-    OUTPUT_VARIABLE GIT_OUTPUT)
-
-  # this may fail on shallow clones that only knows about a limited number of commits.
-  # if there is an older merged revision the head, it may not be available to git.
-  if (NOT GIT_OUTPUT)
-      set(ENTERPRISE_BUILD_REPOSITORY "GIT FAILED TO RETRIEVE THE VERSION - SHALLOW CLONE?")
+    COMMAND ${GIT_EXE} rev-parse --short HEAD
+    OUTPUT_VARIABLE COMMIT_RAW
+  )
+  if (NOT COMMIT_RAW)
+    message(FATAL_ERROR "Can't extract current commit with the command: 'git rev-parse --short HEAD'")
+  endif()
+  
+  string(STRIP ${COMMIT_RAW} COMMIT_SHORT)
+  
+  if (NOT DEFINED BUILD_REPO_INFO OR BUILD_REPO_INFO STREQUAL "default")
+    execute_process(
+      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/enterprise
+      COMMAND ${GIT_EXE} branch --show-current
+      OUTPUT_VARIABLE BRANCH_NAME_RAW)
+    if (NOT BRANCH_NAME_RAW)
+      # For example, in docker we do 'checkout'. Hence, it is impossible to detect branch
+      set(ENTERPRISE_BUILD_REPOSITORY "${COMMIT_SHORT}")
       set(HAVE_ENTERPRISE_BUILD_REPOSITORY "1")
     else()
-      string(STRIP ${GIT_OUTPUT} REPOSITORY_VERSION)
-      set(ENTERPRISE_BUILD_REPOSITORY ${REPOSITORY_VERSION})
+      string(STRIP ${BRANCH_NAME_RAW} BRANCH_NAME)
+      set(ENTERPRISE_BUILD_REPOSITORY "refs/${BRANCH_NAME} ${COMMIT_SHORT}")
       set(HAVE_ENTERPRISE_BUILD_REPOSITORY "1")
     endif()
+  elseif(BUILD_REPO_INFO STREQUAL "release")
+    execute_process(
+      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/enterprise
+      COMMAND ${GIT_EXE} describe --all --tags --match v${ARANGODB_PLAIN_VERSION}
+      OUTPUT_VARIABLE TAG_RAW)
+    if (NOT TAG_RAW)
+      message(FATAL_ERROR "Can't extract tag using the command: 'git describe --all --tags --match v${ARANGODB_PLAIN_VERSION}")
+    else()
+      string(STRIP ${TAG_RAW} TAG)
+      set(ENTERPRISE_BUILD_REPOSITORY "refs/${TAG} ${COMMIT_SHORT}")
+      set(HAVE_ENTERPRISE_BUILD_REPOSITORY "1")
+    endif()
+  elseif(BUILD_REPO_INFO STREQUAL "nightly")
+    set(ENTERPRISE_BUILD_REPOSITORY "refs/head/${ARANGODB_VERSION_MAJOR}.${ARANGODB_VERSION_MINOR} ${COMMIT_SHORT}")
+    set(HAVE_ENTERPRISE_BUILD_REPOSITORY "1")
+  else ()
+    set(ENTERPRISE_BUILD_REPOSITORY "GIT FAILED TO RETRIEVE THE VERSION - UNSUPPORTED BUILD MODE")
+    set(HAVE_ENTERPRISE_BUILD_REPOSITORY "1")
+  endif()
 else ()
   set(ENTERPRISE_BUILD_REPOSITORY "")
   set(HAVE_ENTERPRISE_BUILD_REPOSITORY "0")
```

Comment:
This is essentially a duplication of what we do above for the main repo. Can we keep this DRY and extract it to a separate function?